src/components/common/FilterGroup/filters/FilterMultipleSelection.tsx

```diff
@@ -33,16 +32,33 @@ export default function MultipleSelection(props: Props): JSX.Element {
     setSelections(props.values);
   }, [props.values]);
 
-  const handleChange = (value: string | null) => {
-    const updatesValues = [...selections];
+  const optionsMap = useMemo(
+    () =>
+      new Map<string, string>(
+        props.options.filter((opt) => opt.value && opt.label).map((opt) => [opt.value!, opt.label!])
+      ),
+    [props.options]
+  );
 
+  const onAdd = (value: string | null) => {
+    const updatesValues = [...selections];
     const valueIndex = updatesValues.findIndex((v) => v === value);
     if (valueIndex < 0) {
       updatesValues.push(value);
-    } else {
+    }
+    updateFilters(updatesValues);
+  };
+
+  const onRemove = (value: string | null) => {
+    const updatesValues = [...selections];
+    const valueIndex = updatesValues.findIndex((v) => v === value);
+    if (valueIndex >= 0) {
       updatesValues.splice(valueIndex, 1);
     }
+    updateFilters(updatesValues);
```

Comment:
same, move it into the if block.. i understand this may not really happen (we won't remove something that doesn't exist) ..